config/cmake/HDFCXXCompilerFlags.cmake

```diff
@@ -9,17 +9,45 @@
 # If you do not have access to either file, you may request a copy from
 # help@hdfgroup.org.
 #
+ENABLE_LANGUAGE (CXX)
+
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+
 set(CMAKE_CXX_EXTENSIONS OFF)
 
 set (CMAKE_CXX_FLAGS "${CMAKE_CXX_SANITIZER_FLAGS} ${CMAKE_CXX_FLAGS}")
 if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.15.0")
-  message (VERBOSE "Warnings Configuration: CXX default:  ${CMAKE_CXX_FLAGS}")
+  message (VERBOSE "Warnings Configuration: CXX default: ${CMAKE_CXX_FLAGS}")
 endif ()
 #-----------------------------------------------------------------------------
 # Compiler specific flags : Shouldn't there be compiler tests for these
 #-----------------------------------------------------------------------------
+if(WIN32 AND CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
+  set(_INTEL_WINDOWS 1)
+endif()
+
+if(WIN32 AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang"
+   AND "x${CMAKE_CXX_SIMULATE_ID}" STREQUAL "xMSVC")
+  set(_CLANG_MSVC_WINDOWS 1)
+endif()
+
+# MSVC 14.28 enables C5105, but the Windows SDK 10.0.18362.0 triggers it.
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 19.28)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -wd5105")
+endif()
+
+if (CMAKE_CXX_COMPILER_ID STREQUAL SunPro AND
+    NOT DEFINED CMAKE_CXX${CMAKE_CXX_STANDARD}_STANDARD_COMPILE_OPTION)
+  if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.13)
+    if (NOT CMAKE_CXX_STANDARD OR CMAKE_CXX_STANDARD EQUAL 98)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++03")
```

Comment:
Did you test this? C++11 is required in develop.

Reply:
This section was copied from a CMake file. That line is just setting the min to c++03 if CMAKE_CXX_STANDARD is not defined - we set that in line 14.